### PR TITLE
Emit Actual Message on Input Failure

### PR DESCRIPTION
**Pull request by bska**
This gives more information to the user and hopefully aids them in resolving the underlying issue.

### CODE REVIEW

**Comment by bska** [comment]
When combined with OPM/opm-common#3027, this PR changes the diagnostic message
```
Error: Unrecoverable errors were encountered while loading input
```
into
```
Error: Unrecoverable errors were encountered while loading input: Problem with keyword DATES
In PROJ2_RESTART2.DATA line 408
Keyword is not properly terminated
```
when faced with the underlying issue presented in [#3846](https://github.com/OPM/opm-simulators/issues/3846#issuecomment-1121927690)

**Comment by bska** [comment]
jenkins build this please

**Comment by OPMUSER** [comment]
@bska This is very good, this will fix a ton strange error messages. One thing with the first line of the message, that is:

`Error: Unrecoverable errors were encountered while loading input: Problem with keyword DATES`

Is 92 characters long, do you think we should shorten this to less than or equal to 80 characters for terminal output? For example:

`Error: Unrecoverable errors while loading input: Problem with keyword DATES`

Will this fix still work with multi-line input, for example:

```
--
--       DEFINE MICROBIALLY INDUCED CALCITE PRECIPITATION PARAMETERS
--
--       DEN     DEN     DET     CRI     FIT     HVE     HVE     MGR         
--       BIOF    CAL     RAT     POR     FAC     OXY     URE     RAT         
--       ------  ------  ------  ------  ------  ------  ------  ------  ------ 
--       MOX     MUR     MUR     MIA     MID     MIN     OXC     TOL     YGR
--       CON     CON     UTI     RAT     RAT     PER     FAC     CLO     COE
--       ------  ------  ------  ------  ------  ------  ------  ------  ------   
MICPPARA    
         35.     2710.   2.6E-13 0.1     3.      2E-5    21.3    3.6      
         0.04    60.     1391.   0.074   0.028  1.013E-5 0.5     1E-4    0.5    /

and SGOFLET, SWGFLET and SWOFLET keywords spanning multiple lines. 

```
Just asking because I cannot test.

**Comment by bska** [comment]
> One thing with the first line of the message[...] is 92 characters long, do you think we should shorten this to less than or equal to 80 characters for terminal output?

I do&mdash;that's a really good suggestion.  I've pushed an update to match your suggested alternative.  Incidentally, this update makes the length of the first line 70 + number of characters in the keyword name which will be below 80 unless we have 10-character keyword names at some point.

> Will this fix still work with multi-line input,

Yes.  The detection is based on the terminating `/` character being present or absent.  We do not care about individual lines at this point.

**Comment by bska** [comment]
jenkins build this please

**Comment by blattms** [comment]
thanks a lot really appreciated. Looks great as always. just one more jenkins dance needed.

**Comment by blattms** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please
